data/tribes/buildings/productionsites/atlanteans/well/init.lua

```diff
@@ -43,6 +43,10 @@ wl.Descriptions():new_productionsite_type {
          -- TRANSLATORS: Completed/Skipped/Did not start working because ...
          descname = _("working"),
          actions = {
+            -- time total: 20 + 20 + 3.6 = 43.6 sec
+            -- average time if water resource is depleted:
+            --   (65 * (20 + 20 + 3.6) + 35 * (20 + 20 + 10)) / 65 = 70.523 sec
+            -- number 10 is measured experimentally
```

Comment:
OK, I didn't know this. Now I've removed these additional comments. From my side all is finished in this PR.

Just one question: I've measured time of atlantean gold mine, when there are no gold resources around. Full cycle took about 126.5 sec, measured by food consumption. I expected this time: 40.2 + 3 * (22 + 10) = 136.2 sec. Why are there only two skipped program delays for three mining attempts?

Reply:
I looked up the code, and can confirm it must be the failed/skipped program delay.

So I think I kind of understand now how the delay works in `ProductionSite::program_start()` (and `…::program_end()`). It only prevents the same program to run within 10s after a failure, so if other programs run that take longer than that, then next time the failed program can be tried again immediately.

In the case of the well, the main program itself fails, so it has to wait 10s.

OTOH, in the case of the gold mine, when a subprogram fails, it only gets skipped (according to the documentation of `on failure` in `production_program.cc`), and the main program continues with the next step. So the delay applies after the first and second failed tries, but not after the last one, because of the sleep before the first try in the next run of the main program. AFAICT the main program never returns a failure because of a subprogram, see `ProductionProgram::ActCall::execute()`, so there's no delay for the main program here.

(BTW this also means that it will be rather painful to calculate the average times of empty mines with the same programs (hmmm, `avg_delay = 10*(n_calls - 1)/n_calls`?)… and even more so for the atl. crystal mine and emp. marble mine…)